### PR TITLE
Adding support for changing existing managed instance hardware generation.

### PR DESCRIPTION
**Pull request by v-nestan**
## Description

This change adds support for changing existing managed instance hardware generation.
It can be changed using SkuName and ComputeGeneration parameters.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
image

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by msJinLei** [comment]
@v-nestan Please fix the build error
````
2019-12-31T05:27:12.8145904Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: Invalid Resx file. System.Xml.XmlException: The 'data' start tag on line 601 position 4 does not match the end tag of 'root'. Line 606, position 3. [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8146996Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReaderImpl.Throw(Exception e) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8147296Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReaderImpl.Throw(String res, String[] args) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8147440Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReaderImpl.ThrowTagMismatch(NodeData startTag) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8147879Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReaderImpl.ParseEndElement() [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8148083Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReaderImpl.ParseElementContent() [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8148257Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.XmlTextReader.Read() [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8148603Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8148783Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r, LoadOptions o) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8148990Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
2019-12-31T05:27:12.8149173Z D:\a\1\s\src\Sql\Sql\Properties\Resources.resx : error MSB3103: at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.ReadResources(Stream s, String filename, Boolean pathsRelativeToBasePath) [D:\a\1\s\src\Sql\Sql\Sql.csproj] [D:\a\1\s\build.proj]
````

**Comment by isra-fel** [comment]
Kindliy ping @v-nestan

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ## Version 2.1.1
 * Update references in .psd1 to use relative path
 * Upgraded storage creation in Vulnerability Assessment auto enablement to StorageV2
+* Adding support for changing existing managed instance hardware generation
```
Comment (msJinLei) [inline src/Sql/Sql/ChangeLog.md]: Please rebase the master and add this new changelog line understand upcoming release

File: src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs [hunk 2]
```diff
@@ -246,6 +420,7 @@ protected override IEnumerable<AzureSqlManagedInstanceModel> PersistChanges(IEnu
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            // System.Diagnostics.Debugger.Launch();
```
Comment (msJinLei) [inline src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs]: remove the commented line.

**Comment by isra-fel** [comment]
Closing for lack of response, and lots of conflicting files. Please feel free to reopen to continue. Thanks

**Pull request closed without merging**